libraries/FunctionalInterrupt/examples/Functional/Functional.ino

```diff
@@ -0,0 +1,75 @@
+// Hint: On ESP8266, this no longer depends on FunctionalInterrupt, but is rather a feature of
+// the standard API in Arduino.h
+
+//#include <FunctionalInterrupt.h>
+
+#ifndef IRAM_ATTR
+#define IRAM_ATTR ICACHE_RAM_ATTR
+#endif
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      //attachInterruptArg(PIN, reinterpret_cast<void(*)(void*)>(&buttonIsr_static), this, FALLING);
+      attachInterrupt(PIN, std::bind(&Button::buttonIsr, this), FALLING);
+    };
+    ~Button() {
+      detachInterrupt(PIN);
+    }
+
+    void IRAM_ATTR buttonIsr() {
+      numberKeyPresses += 1;
+      pressed = true;
+    }
+
+    static void IRAM_ATTR buttonIsr_static(Button* const self) {
+      self->buttonIsr();
+    }
+
+    uint32_t checkPressed() {
+      if (pressed) {
+        Serial.printf("Button on pin %u has been pressed %u times\n", PIN, numberKeyPresses);
+        pressed = false;
+      }
+      return numberKeyPresses;
+    }
+
+  private:
+    const uint8_t PIN;
+    volatile uint32_t numberKeyPresses = 0;
+    volatile bool pressed = false;
+};
+
+Button* button1;
+Button* button2;
+
+
+void setup() {
+  Serial.begin(115200);
+  Serial.println("FunctionalInterrupt test/example");
+
+  button1 = new Button(BUTTON1);
```

Comment:
I suspect you're doing this, because a global instance doesn't work (can't attach in the constructor). Please do one of the following:
- change the class implementation to follow Arduino patterns, e.g.: add begin() and end() methods that do the actual hw init and release.
- implement a default constructor that does nothing, and an assignment constructor that does the same as the current pin constructor. Then, instantiate globally with default, and here do an assignment of a temp object built with the pin constructor
- add a comment here explaining why you're doing a new, that this is ok only in setup because the mem is never released

Reply:
Opted for last in your list.